js/apiv2.js

```diff
@@ -130,16 +130,42 @@ const api = {
             })
             .catch((error) => {
                 this.logFailure(error)
-                return error;
+                return {error};
             })
     },
     async uploadTemplate(data) {
         // TODO
-            // return this.templatesInstance.put('/', data)
+            // Send to backend
         console.log("API request to send Templates");
+            return await Promise.resolve( {statusText: "OK", data: {valid: 16, invalid: 0}} )
+                .then(response => new Promise( resolve => {
+                    let timeOptions = [500, 900, 1000, 2000, 3000]
+                    let delay = timeOptions[Math.floor(Math.random() * 5)]
```

Comment:
Add a comment about what this is for?

Reply:
I added some comments. This is temporary code to mock api responses at different time delays to see how the loading spinner handles it. I'll replace all this with the actually call when its ready.